src/jet_client.rs

```diff
@@ -144,224 +151,331 @@ impl Future for JetMsgReader {
+                                    candidate.set_state(CandidateState::Accepted);
+
+                                    status_code = StatusCode::OK;
                                 }
+                            } else {
+                                status_code = StatusCode::NOT_FOUND;
                             }
                         }
-
-                        self.response_msg = Some(JetMessage::JetAcceptRsp(JetAcceptRsp {
-                            status_code,
-                            version: request.version,
-                            association: Uuid::nil(),
-                            instance: self.config.jet_instance(),
-                            timeout: ACCEPT_REQUEST_TIMEOUT_SEC,
-                        }));
-                    }
-                    _ => {
-                        // No jet message exist with version different than 1 or 2
-                        unreachable!()
                     }
+
+                    (status_code, Uuid::nil())
                 }
-            } else {
-                return Ok(Async::NotReady);
+                _ => {
+                    // No jet message exist with version different than 1 or 2
+                    unreachable!()
+                }
+            };
+
+            if request.version == 1 && status_code == StatusCode::OK {
+                self.remove_association_future = Some(Box::new(create_remove_association_future(
+                    self.jet_associations.clone(),
+                    association.clone(),
+                )));
             }
+
+            // Build response
+            let response_msg = JetMessage::JetAcceptRsp(JetAcceptRsp {
+                status_code,
+                version: request.version,
+                association,
+                instance: self.config.jet_instance().clone(),
+                timeout: ACCEPT_REQUEST_TIMEOUT_SEC,
+            });
+            let mut response_msg_buffer = Vec::with_capacity(512);
+            response_msg.write_to(&mut response_msg_buffer)?;
+
+            Ok(Async::Ready(response_msg_buffer))
+        } else {
+            Ok(Async::NotReady)
         }
+    }
 
-        // We have a response for sure ==> Send response
-        let response_msg = self.response_msg.as_ref().expect("We must have a response to send");
-        let mut v = Vec::new();
-        response_msg.write_to(&mut v)?;
-        try_ready!(self.transport.poll_write(&v));
-
-        // Start timeout to remove the association if no connect is received. We start it only if a new association has just been added,
-        // possible only with version 1.
-        if let JetMessage::JetAcceptRsp(accept_rsp) = response_msg {
-            if accept_rsp.version == 1 && accept_rsp.status_code == StatusCode::OK {
-                start_remove_association_future(self.executor_handle.clone(), self.jet_associations.clone(), accept_rsp.association);
+    fn handle_set_transport(&mut self) -> Poll<(), io::Error> {
```

Comment:
After we have written the response, depending on the version, we set up the transport